internal/pkg/cli/deploy/workload.go

```diff
@@ -360,6 +368,38 @@ func (d *workloadDeployer) generateCloudFormationTemplate(conf stackSerializer)
 	}, nil
 }
 
+func (d *workloadDeployer) deployAndHandleInterrupt(conf cloudformation.StackConfiguration, opts []awscloudformation.StackOption) (DeployWorkloadOutput, error) {
+	g, ctx := errgroup.WithContext(context.Background())
+	ctx, cancel := context.WithCancel(ctx)
+	defer cancel()
+	out := DeployWorkloadOutput{}
+	deployService := func() error {
+		defer cancel()
+		err := d.deployer.DeployService(ctx, conf, d.resources.S3Bucket, opts...)
+		if err != nil {
+			if !errors.Is(err, context.Canceled) {
+				return err
+			}
+		}
+		return nil
+	}
+	waitForSignalAndHandleInterrupt := func() error {
+		isDeleted, isUpdateCanceled, err := d.deployer.WaitForSignalAndHandleInterrupt(ctx, cancel, conf.StackName())
+		out.IsWkldDeleted = isDeleted
+		out.IsWkldUpdateCanceled = isUpdateCanceled
+		if err != nil {
+			return err
+		}
+		return nil
```

Comment:
I have a question 🤚🏼 , not a request! Why do you prefer putting `WaitForSignal` in `cloudformation` package? An alternative I can think of is having the "wait" logic here:

```go
// Here
waitForSignal := func() error {
    sigInt  ch os.Signal
    signal.Notify(sigInt, os.SIGINT)
    select {
		case <-sigInt:
                    log.Infof("Received SIGINT.....")
                    cancel()
                    err := d.deployer.CancelUpdate()
                 case <- ctx.Done():
                     signal.Stop(sigInt)
                     close(sigInt)
     }
}
```

Reply:
If I want to enable to `ctrl-c` for `env deploy`  then I think 
1. I need to replicate the same logic for the env deployer.
2. And also in order to know the status of the cloudformation stack, I need to public method `DescribeStack` that i can call in `d,deployer.DescribeStack`.

Reply:
> I need to replicate the same logic for the env deployer.

I think arguably, this can be mitigated by ⬇️ 
```go
function waitForSignal(handler func() error ) error {
     case <-sigInt:
            log.Infof("Received SIGINT.....")
            cancel()
            if err := handler(); err != nil {...}
   case <- ctx.Done():
       signal.Stop(sigInt)
       close(sigInt)
}

// In workload deployer 
g.Go(deployService)
g.Go(waitForSignal(func() error {
    err := d.deployer.CancelUpdate()
}))

// In env deployer 
g.Go(deployEnv)
g.Go(waitForSignal(func() error {
    err := d.deployer.CancelEnvironmentUpdate()
}))
```

> And also in order to know the status of the cloudformation stack, I need to have public method DescribeStack that i can call d.deployer.DescribeStack.'

This can & probably should be inside of `CancelUpdate` right?
```go
// In deploy/cloudformation/cloudformation.go
func (cf CloudFormation) CancelUpdate() error {
   stackDescr, err := cf.cfnClient.DescribeStack()
   switch  aws.StringValue(stackDescr.StackStatus) {
   case sdkcloudformation.StackStatusCreateInProgress:
   case sdkcloudformation.StackStatusUpdateInProgress 
   }
}
```

Reply:
This is really good. Yeah I think this is a better way of having the abstractions than what i have before.